File: src/main/java/com/rj/ecommerce_email_service/contract/v1/ShippingMethod.java
package com.rj.ecommerce_email_service.contract.v1;

public enum ShippingMethod {
    INPOST,
    DHL;

    public static ShippingMethod fromString(String shippingMethod) {
        try {
            return ShippingMethod.valueOf(shippingMethod);
        } catch (IllegalArgumentException e) {
            throw new IllegalArgumentException("Invalid shipping method: " + shippingMethod, e);
        }
    }
}

File: shared-api/src/main/kotlin/com/rj/ecommerce/api/shared/enums/ShippingMethod.kt
package com.rj.ecommerce.api.shared.enums

/**
 * Represents the method used for shipping.
 *
 * Values:
 * - INPOST: Shipping via InPost (parcel lockers)
 * - DHL: Shipping via DHL courier
 * - STANDARD: Standard shipping method
 * - EXPRESS: Express shipping method
 */
enum class ShippingMethod {
    INPOST,
    DHL,
    STANDARD,
    EXPRESS
}
